components/create_team/components/team_url/team_url.tsx

```diff
@@ -2,51 +2,62 @@
 // See LICENSE.txt for license information.
 /* eslint-disable react/no-string-refs */
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Button, Tooltip} from 'react-bootstrap';
 import ReactDOM from 'react-dom';
 import {FormattedMessage} from 'react-intl';
 
+import {ActionFunc} from 'mattermost-redux/types/actions';
+import {Team} from 'mattermost-redux/types/teams';
+
 import {trackEvent} from 'actions/diagnostics_actions.jsx';
+
 import Constants from 'utils/constants.jsx';
 import * as URL from 'utils/url';
 import logoImage from 'images/logo.png';
 
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 import OverlayTrigger from 'components/overlay_trigger';
 
-export default class TeamUrl extends React.PureComponent {
-    static propTypes = {
+type State = {
+    isLoading: boolean;
+    nameError: string | JSX.Element;
+}
+
+type Props = {
 
-        /*
-         * Object containing team's display_name and name
-         */
-        state: PropTypes.object,
+    /*
+     * Object containing team's display_name and name
+     */
+    state: {team: any; wizard: string};
+
+    /*
+     * Function that updates parent component with state props
+     */
+    updateParent: (state: Props['state']) => void;
+
+    /*
+     * Object with redux action creators
+     */
+    actions: {
 
         /*
-         * Function that updates parent component with state props
+         * Action creator to check if a team already exists
          */
-        updateParent: PropTypes.func,
+        checkIfTeamExists: (teamName: string) => ActionFunc;
 
         /*
-         * Object with redux action creators
-         */
-        actions: PropTypes.shape({
-
-            /*
-             * Action creator to check if a team already exists
-             */
-            checkIfTeamExists: PropTypes.func.isRequired,
-
-            /*
-             * Action creator to create a new team
-             */
-            createTeam: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+     * Action creator to create a new team
+     */
+        createTeam: (team: Team) => ActionFunc;
+    };
+    history: {
+        push(path: string): void;
+    };
+}
 
-    constructor(props) {
+export default class TeamUrl extends React.PureComponent<Props, State> {
+    public constructor(props: Props) {
```

Comment:
Is there a specific reason this is `public`? Usually we're not creating new instances of this component in JS/TS.

Reply:
No, there isn't any specific reason. I'll remove it.